tests/v2/integration/setup/main.go

```diff
@@ -2,116 +2,108 @@ package main
-	"github.com/rancher/rancher/tests/framework/clients/k3d"
+	provisioningv1api "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
 	rancherClient "github.com/rancher/rancher/tests/framework/clients/rancher"
 	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
-	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
 	"github.com/rancher/rancher/tests/framework/extensions/token"
 	"github.com/rancher/rancher/tests/framework/pkg/config"
 	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
-	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2prov/clients"
+	"github.com/rancher/rancher/tests/v2prov/cluster"
+	testdefaults "github.com/rancher/rancher/tests/v2prov/defaults"
 	"github.com/sirupsen/logrus"
-	"k8s.io/client-go/util/retry"
-)
-
-var (
-	agentTag         = os.Getenv("AGENT_TAG")
-	masterAgentImage = "rancher/rancher-agent:" + agentTag
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
-	k3dClusterNameBasename = "k3d-cluster"
+	clusterNameBaseName = "integration-test-cluster"
 )
 
-// setup for integration testing
+// main creates a test namespace and cluster for use in integration tests.
 func main() {
-	rancherConfig := new(rancherClient.Config)
-
-	user := &management.User{
-		Username: "admin",
-		Password: "admin",
+	logrus.Infof("Generating test config")
+	ipAddress, err := getOutboundIP()
+	if err != nil {
+		logrus.Fatalf("Error getting outbound IP address: %v", err)
 	}
 
-	logrus.Infof("Generating test config...")
-	ipAddress := getOutboundIP()
 	hostURL := fmt.Sprintf("%s:8443", ipAddress.String())
-	token, err := token.GenerateUserToken(user, hostURL)
+	userToken, err := token.GenerateUserToken(
+		&management.User{
+			Username: "admin",
+			Password: "admin",
+		},
+		hostURL,
+	)
 	if err != nil {
-		logrus.Fatalf("error with generating admin token: %v", err)
+		logrus.Fatalf("Error with generating admin token: %v", err)
 	}
 
-	clusterName := namegen.AppendRandomString(k3dClusterNameBasename)
-
 	cleanup := true
-	rancherConfig.AdminToken = token.Token
-	rancherConfig.Host = hostURL
-	rancherConfig.Cleanup = &cleanup
-	rancherConfig.ClusterName = clusterName
-
-	if err := defaults.Set(rancherConfig); err != nil {
-		logrus.Fatalf("error with setting up config file: %v", err)
+	rancherConfig := rancherClient.Config{
+		AdminToken:  userToken.Token,
+		Host:        hostURL,
+		Cleanup:     &cleanup,
+		ClusterName: namegen.AppendRandomString(clusterNameBaseName),
 	}
 
-	config.WriteConfig(rancherClient.ConfigurationFileKey, rancherConfig)
-
-	logrus.Infof("Setting up K3D downstream cluster...")
-	testSession := session.NewSession()
-
-	client, err := rancherClient.NewClient("", testSession)
+	err = defaults.Set(&rancherConfig)
 	if err != nil {
-		logrus.Fatalf("error creating admin client: %v", err)
+		logrus.Fatalf("Error with setting up config file: %v", err)
```

Comment:
also one nit pick it's go standards to have errors begin in lower case.

Reply:
@igomez06 , this standard is something that's often misinterpreted - it only applies to actual error messages and not to text that gets logged, https://github.com/go-lang-plugin-org/go-lang-idea-plugin/issues/2151 has a good summary on this.

Reply:
@snasovich sounds good, thanks for the summary! :grin: